README.md

```diff
@@ -10,32 +10,275 @@
+* Optional parameters need not be all present when executing a command, they are denoted in this user guide using square brackets '[]'.<br>
+  e.g. in `edit task m/<module name> ti/<task id> [a/<edited task name>] [d/<edited task deadline>]`, `a/<edited task name>` and `d/<edited task deadline>` are the optional parameters
 
-#### Module delete student
+</div>
 
-* Delete a student from a specific module
-* Format: `module delete <module name>, <student id>`
 
-#### Module delete
+### Add a new module: `add module`
 
-* Delete a module
-* Format: `module delete <module name>`
+Add a new module to TAB.<br>
+Module name must be an NUS module code, consisting of 2-4 capital letters, followed by 4 numbers, followed by 0-4 capital letters.
+Format: `add module m/<module name>`
+
+Examples:
+* `add module m/CS2103`: Add module CS2103 to TAB
+* `add module m/CS2100`: Add module CS2100 to TAB
+
+### Edit a module's name: `edit module`
+
+Edit the name of the module.<br>
+Module name must be an NUS module code, consisting of 2-4 capital letters, followed by 4 numbers, followed by 0-4 capital letters.
+Format: `edit module m/<old module name> mn/<new module name>`
+
+Examples:
+* `edit module m/CS2103 mn/CS2105`:
+    * Edit the module name of module "CS2103" to "CS2105"
+
+### Delete a module from TAB: `delete module`
+
+Delete the specified module from TAB.
+Module name must be an NUS module code, consisting of 2-4 capital letters, followed by 4 numbers, followed by 0-4 capital letters.
+Format: `delete module m/<module name>`
+
+Examples:
+* `delete module m/CS2103`:
+    * Delete module CS2103 from TAB
+* `delete module m/CS2100`:
+    * Delete module CS2100 from TAB
+
+### Add a student to a specific module: `add student`
+
+Add a new student to the specified module.<br>
+Also automatically copies over the task list from the module to the student.<br>
+Student id must be a valid NUS student id (starts with A, followed by 7 numbers and ends with a capital letter.<br>
+Student name should only contain alphanumeric characters and spaces, and it should not be blank. <br>
+Student tele handle must start with @, and followed by 5-32 characters containing only alphanumeric characters and the underscore '_'.<br>
+Student email must follow the format specified in the [Parameter summary](#parameter-summary-with-prefix-included) section below.<br>
+Student id, student tele handle, and student email must be unique for each student.<br>
+Duplicate student names are allowed to cater for students who may have the same name.
+
+Format: `add student m/<module name> i/<student id> n/<student name> t/<student tele handle> e/<student email>`
+
+Examples:
+* `add student m/CS2103 i/A0123456A n/John Doe t/@johndoe e/john.doe@u.nus.edu`:
+    * Add student with id A0123456A to module CS2103
+* `add student m/CS2100 i/A0123457A n/Amy t/@amylee e/amy@u.nus.edu`:
+    * Add student with id A0123457A to module CS2100
+
+### Edit a student's information: `edit student`
```

Comment:
I think if we are making students to have their own identities, we shouldn't include module name as a parameter here, and we can say something like "editing a student will edit his/her information in all modules"

Reply:
Agreed. Will update the description.

Reply:
If we are removing the module param then we will need to update the parser and command as well. Should open an issue to discuss this